src/control/describeIndex.ts

```diff
@@ -0,0 +1,76 @@
+import { Static, Type } from '@sinclair/typebox';
+import { IndexOperationsApi } from '../pinecone-generated-ts-fetch';
+import { buildValidator } from '../validator';
+import { mapHttpStatusError, type FailedRequestInfo } from '../errors';
+import type {
+  ResponseError,
+  IndexMeta as IndexDescription,
+} from '../pinecone-generated-ts-fetch';
+
+// If user passes the empty string for index name, the generated
+// OpenAPI client will call /databases/ which is the list
+// indexes endpoint. This returns 200 instead of 404, but obviously
+// no descriptive information is returned for an index named empty
+// string. To avoid this confusing case, we require lenth > 1.
+const DescribeIndexOptionsSchema = Type.String({ minLength: 1 });
```

Comment:
This is very useful context for me to have, but for now I think I would prefer to leave other forms of index name validation to the API so that clients can continue to work even if the API/backend requirements change (since they are the ultimate source of truth). I only added this >1 length requirement basically to guard against janky behavior in the generated openapi client when the index name length is zero being equivalent to calling the list indexes endpoint and displaying nonsense results.